### PR TITLE
Join lobby

### PR DESCRIPTION
**Pull request by Zephyr75**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java [hunk 1]
```diff
@@ -32,7 +32,8 @@ public class PlanthuntCameraActivity extends AppCompatActivity {
 
     private static Button collectionButton;
     private static Button captureButton;
-    private static Button lobbyButton;
+    private static Button createLobbyButton;
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCameraActivity.java:35]: As the codeclimate warning says, you should not make those variables as static (all the static fields in this class). Maybe you meant them to be final I assume ?

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.Objects;
+
+import ch.epfl.sdp.healthplay.database.Database;
+import ch.epfl.sdp.healthplay.database.Lobby;
+
+public class PlanthuntJoinLobbyActivity extends AppCompatActivity {
+
+    private FirebaseUser user;
+    private static Button lobbyButton;
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java:24]: Same as the other comment, you should not make this field a static field as this is an Android context class and does a memory leak as the codeclimat warning says.

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 3]
```diff
@@ -307,17 +308,27 @@ public void writeNewLobby (String name, String password, String hostUid, int rem
...
+                int nbrPlayers = Integer.parseInt(dataSnapshot.getValue().toString()) + 1;
+                mDatabase
+                        .child(LOBBIES)
+                        .child(name)
+                        .child(NBR_PLAYER)
+                        .setValue(nbrPlayers);
+                mDatabase
+                        .child(LOBBIES)
+                        .child(name)
+                        .child("playerUid" + (nbrPlayers))
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:328]: You could make a constant for this string in here.

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 4]
```diff
@@ -362,4 +373,18 @@ public void onSuccess(DataSnapshot dataSnapshot) {
...
+     * Updates the score of a player in the lobby
+     *
+     * @param name      the unique identifier given to the lobby
+     * @param password  the unique password given to the lobby
+     */
+    public Task checkLobbyId (String name, String password){
+        return mDatabase
+                .child(LOBBIES)
+                .child(name)
+                .child("password")
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:387]: You could make a constant for this string.

**Review by ozan-guven — CHANGES_REQUESTED** [review]
Nice implementation. You can look for my comments.

**Review by ozan-guven — APPROVED** [review]
Nice job ! You didn't need to completely remove the static field variables, you could have just removed the `static` keyword but this also works !

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCreateLobbyActivity.java [hunk 5]
```diff
@@ -41,17 +40,18 @@ public void onClick(View v) {
...
-                db.addUserToLobby(newLobby.getName(), newLobby.getNbrPlayers(), "testPlayer1");
+                /*db.addUserToLobby(newLobby.getName(), newLobby.getNbrPlayers(), "testPlayer1");
                 newLobby.addPlayer();
                 db.addUserToLobby(newLobby.getName(), newLobby.getNbrPlayers(), "testPlayer2");
                 newLobby.addPlayer();
                 db.updateLobbyTime(newLobby.getName(), TEST_2);
                 db.updateLobbyPlayerScore(newLobby.getName(), uid, TEST_3);
-                db.updateLobbyPlayerScore(newLobby.getName(), "whatever", TEST_4);
+                db.updateLobbyPlayerScore(newLobby.getName(), "whatever", TEST_4);*/
             }
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntCreateLobbyActivity.java:55]: Try to remove comments when creating a PR

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+import java.util.Objects;
+
+import ch.epfl.sdp.healthplay.database.Database;
+import ch.epfl.sdp.healthplay.database.Lobby;
+
+public class PlanthuntJoinLobbyActivity extends AppCompatActivity {
+
+    //private static final int TEST_1 = 180, TEST_2 = 123, TEST_3 = 789, TEST_4 = 456;
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java:24]: Remove comment

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        //Create new lobby when clicking on Create lobby button
+        lobbyButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //Get Strings from input fields
+                String name = editName.getText().toString();
+                String password = editPassword.getText().toString();
+                String username = editUsername.getText().toString();
+
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java:47]: You could create a function with an EditText in parameter, in order not to duplicate code

File: app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                            db.addUserToLobby(name, username);
+                        }
+                    }
+                });
+                /*db.addUserToLobby(newLobby.getName(), newLobby.getNbrPlayers(), "testPlayer2");
+                newLobby.addPlayer();
+                db.updateLobbyTime(newLobby.getName(), TEST_2);
+                db.updateLobbyPlayerScore(newLobby.getName(), uid, TEST_3);
+                db.updateLobbyPlayerScore(newLobby.getName(), "whatever", TEST_4);*/
+            }
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/PlanthuntJoinLobbyActivity.java:64]: Commented line need to be removed

File: app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java [hunk 9]
```diff
@@ -341,25 +355,39 @@ public void updateLobbyTime (String name,int remainingTime){
...
         }
     }
+
+    /**
+     * Updates the score of a player in the lobby
+     *
+     * @param name      the unique identifier given to the lobby
+     * @param password  the unique password given to the lobby
+     */
+    public Task checkLobbyId (String name, String password){
```
Comment (Tetardfurieux) [inline app/src/main/java/ch/epfl/sdp/healthplay/database/Database.java:386]: Wrong description of the function, please fix

**Review by Tetardfurieux — CHANGES_REQUESTED** [review]
Good code, good feature and good use of Objects.requireNonNull on DataSnapshot.
Also good documentation of the code.
However, please cleanup your code when creating a PR, + update the description of the function getLobbyId

**Review by Tetardfurieux — APPROVED** [review]
That's perfect now !

**Pull request merged**